tests/framework/artifacts.py

```diff
@@ -463,11 +464,13 @@ def kernels(self, keyword=None):
             keyword=keyword,
         )
 
-        valid_kernels = list(
-            filter(
-                lambda kernel: any(s in kernel.key for s in SUPPORTED_KERNELS), kernels
-            )
-        )
+        def check_kernel(kernel):
+            kernel_name = re.search("vmlinux-(.*).bin", kernel.key)
+            if kernel_name:
+                return kernel_name.group(1) in SUPPORTED_KERNELS
+            return False
+
+        valid_kernels = list(filter(check_kernel, kernels))
```

Comment:
This tranforms the check so that they have to be exact matches. In that case, this is simpler and doesn't require regexes:

```python
supported_kernels = {f"vmlinux-{sup}.bin" for sup in SUPPORTED_KERNELS}
valid_kernels = supported_kernels.intersection(kernels)
```

Reply:
Well, first of all, `kernels` - is a list of objects, not strings. Second `kernel.key` is a path like `ci-artifacts/kernels/aarch64/vmlinux-4.14.bin`. So actually we can extract last item of the path with `os.path.basename(kernel.key)`.
So it would look like:
```python
supported_kernels = {f"vmlinux-{sup}.bin" for sup in SUPPORTED_KERNELS}
valid_kernels = list(filter(lambda kernel: os.path.basename(kernel.key) in supported_kernels, kernels))
```
Looks better?

Reply:
God point. But list comprehensions are more readable/idiomatic in Python than list+filter. Try

```python
valid_kernels = [kernel for kernel in kernels if Path(kernel.key).name in supported_kernels]
```

Reply:
I see... snakes in snake language